arch/arm/src/sama5/Kconfig

```
@@ -1028,6 +1043,220 @@ config SAMA5_DBGU_PARITY
 endif # !SAMA5_DBGU_NOCONFIG
 endmenu #DBGU Configuration
 
+if SAMA5D2_CLASSD
+
+menu "Class D Peripheral Configuration"
+
+menu "Class D DMA options"
+
+config SAMA5D2_CLASSD_DMA_DMAC_NUMBER
+	int "Class D DMA Controller number"
+	default 0
+	range 0 1
+	---help---
+		Select which xdma controller to use for Class D DMA.
+
+config SAMA5D2_CLASSD_DMADEBUG
+	bool "Class D DMA transfer debug"
+	depends on DEBUG_FEATURES && DEBUG_DMA && DEBUG_DMA_INFO
+	default n
+	---help---
+		Enable special debug instrumentation analyze Class D DMA data transfers.
+		This logic is as non-invasive as possible:  It samples DMA
+		registers at key points in the data transfer and then dumps all of
+		the registers at the end of the transfer.
+
+endmenu # "Class D DMA options"
+
+config SAMA5D2_CLASSD_INFLIGHT
+	int "SAMA5 Class D maximum in-flight audio buffers"
+	default 2
+
+config SAMA5D2_CLASSD_PAD_CLK
+	bool "Enable Audio Clock Output pin"
+	default n
+
+config SAMA5D2_CLASSD_LEN
+	bool "Left Channel Enable (LEN)"
+	default y
+
+config SAMA5D2_CLASSD_REN
+	bool "Right Channel Enable (REN)"
+	default y
+
+choice
+	prompt "Class D PWM Modulation Type"
+	default SAMA5D2_CLASSD_PWM_TRAILING
+	---help---
+		TRAILING EDGE: The signal is single-ended.
+			If bit NON_OVERLAP is cleared, the signal is sent to 
+			CLASSD_L0 and CLASSD_R0.
+
+			If bit NON_OVERLAP is set, the signal is sent to
+			CLASSD_L0/L1 and CLASSD_R0/R1.
+		UNIFORM: The signal is differential.
+			If bit NON_OVERLAP is cleared, the signal is sent to 
+			CLASSD_L0/L2 and CLASSD_R0/R2.
+			
+			If bit NON_OVERLAP is set, the signal is sent to 
+			CLASSD_L0/L1/L2/L3 and CLASSD_R0/R1/R2/R3.
+
+config SAMA5D2_CLASSD_PWM_TRAILING
+bool "Trailing"
+
+config SAMA5D2_CLASSD_PWM_UNIFORM
+bool "Uniform"
+
+endchoice # Class D PWM Modulation Type
+		
+config SAMA5D2_CLASSD_NON_OVERLAP_MODE
+	bool "Non-overlapping Circuit Enable (NON_OVERLAP)"
+	default n
+
+choice
+	depends on SAMA5D2_CLASSD_NON_OVERLAP_MODE
+	prompt "Non-overlapping Value (NOVRVAL)"
+	default SAMA5D2_CLASSD_NON_OVERLAP_5NS
+	---help---	
+		Non-overlapping time
+
+config SAMA5D2_CLASSD_NON_OVERLAP_5NS
+bool "5ns"
+
+config SAMA5D2_CLASSD_NON_OVERLAP_10NS
+bool "10ns"
+
+config SAMA5D2_CLASSD_NON_OVERLAP_15NS
+bool "15ns"
+
+config SAMA5D2_CLASSD_NON_OVERLAP_20NS
+bool "20ns"
```

Seems I made a wrong suggestion last time
```suggestion
config SAMA5D2_CLASSD_NON_OVERLAP_5NS
	bool "5ns"

config SAMA5D2_CLASSD_NON_OVERLAP_10NS
	bool "10ns"

config SAMA5D2_CLASSD_NON_OVERLAP_15NS
	bool "15ns"

config SAMA5D2_CLASSD_NON_OVERLAP_20NS
	bool "20ns"
```